integration/read_write_mode_test.go

```diff
@@ -154,18 +174,44 @@ func TestReadWriteModeRecordingRule(t *testing.T) {
 	// Wait for recording rule to evaluate
 	require.NoError(t, cluster.backendInstance.WaitSumMetricsWithOptions(e2e.GreaterOrEqual(1), []string{"cortex_prometheus_rule_evaluations_total"}, e2e.WaitMissingMetrics))
 
+	runRecordingRuleQuery(t, client, testRuleNameFloat, seriesFloat)
+	runRecordingRuleQuery(t, client, testRuleNameHisto, seriesHisto)
+}
+
+func runRecordingRulePush(t *testing.T, client *e2emimir.Client, seriesName string, genSeries generateSeriesFunc) []prompb.TimeSeries {
+	// Push data that should be captured by the recording rule
+	pushTime := time.Now()
+	series, _, _ := genSeries(seriesName, pushTime, prompb.Label{Name: "foo", Value: "bar"})
+
+	res, err := client.Push(series)
+	require.NoError(t, err)
+	require.Equal(t, 200, res.StatusCode)
+
+	return series
+}
+
+func runRecordingRuleQuery(t *testing.T, client *e2emimir.Client, testRuleName string, series []prompb.TimeSeries) {
 	// Verify recorded series is as expected
 	queryTime := time.Now()
 	queryResult, err := client.Query(testRuleName, queryTime)
 	require.NoError(t, err)
 	require.Equal(t, model.ValVector, queryResult.Type())
 
+	metric := model.Metric{
+		labels.MetricName: model.LabelValue(testRuleName),
+	}
+	var sum float64
+	if len(series[0].Samples) > 0 {
+		sum = series[0].Samples[0].Value
+	} else {
+		sum = series[0].Histograms[0].Sum
+		metric["foo"] = model.LabelValue("bar") // TODO(histograms): why is this necessary?
```

Comment:
I feel like we should not have this question and indeed the code in ruler might be wrong?

Reply:
The problem was that histogram_sum which we were using for histograms instead of sum behaves differently with labels since it isn't actually an aggregation function. Changed this part to use sum instead.